.github/workflows/scanner-build.yaml

```diff
@@ -90,6 +90,11 @@ jobs:
         matrix.name == 'prerelease'
       run: echo "GOTAGS=release" >> "$GITHUB_ENV"
 
+    # TODO: remove this step once we can successfully build with CGO for non-x86.
+    - name: Disable CGO for unsupported architectures
+      if: matrix.goarch != 'amd64'
+      run: echo "CGO_ENABLED=0" >> "$GITHUB_ENV"
```

Comment:
This takes care of CI, what about local dev building?

I looked at <https://pkg.go.dev/cmd/cgo>, and it says:

> The cgo tool is enabled by default for native builds on systems where it is expected to work. It is disabled by default when cross-compiling as well as when the CC environment variable is unset and the default C compiler (typically gcc or clang) cannot be found on the system PATH.

So non-x86_64 (e.g., Mac) will have the default behavior overwritten now that we set `CGO_ENABLED`, and things might break. Should we expect people to `export CGO_ENABLED` (or always call `make CGO_ENABLED` to select an appropriate value for them?

Reply:
I believe we only want `CGO_ENABLED` set for downstream or local build on demand, but not for CI nor by default. Is this PR now making it the default?

Reply:
@jvdm My plan was to ask someone with Mac try build scanner (without any exports) but after I get green CI. My plan was to ping you for re-review after that.

@RTann welcome to the party. Yes, you read it correctly that CGO_ENABLED becomes 1 by default for scanner v4 as well. Is there any reason not to do that, like any specific problems this may cause?